cli/src/ui.rs

```diff
@@ -343,6 +343,53 @@ impl Ui {
         Ok(buf)
     }
 
+    /// Repeat the given prompt until the input is one of the specified choices.
+    pub fn prompt_choice(
+        &mut self,
+        prompt: &str,
+        choices: &[&str],
+        default: Option<&str>,
+    ) -> io::Result<String> {
+        if !io::stdout().is_terminal() {
+            if let Some(default) = default {
+                // Choose the default automatically without waiting.
+                writeln!(self.stdout(), "{prompt}: {default}")?;
+                return Ok(default.to_owned());
+            }
+        }
+
+        loop {
+            let choice = self.prompt(prompt)?.trim().to_owned();
```

Comment:
better to terminate on EOF (empty string without `\n`) by mapping to io::Error?

Reply:
Perhaps, EOF should be handled by `prompt()`, and the trailing `"\n"` should be removed there. I have no idea how libgit2 callback handles "{username}\n".

Reply:
Done, added a new commit for this at the base.